File: demo/src/main/java/br/com/instachat/emojikeyboard/adapter/EmojiTabAdapter.java
package br.com.instachat.emojikeyboard.adapter;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.view.ViewGroup;

import br.com.instachat.emojikeyboard.fragments.FragmentEmojiNature;
import br.com.instachat.emojikeyboard.fragments.FragmentEmojiObjects;
import br.com.instachat.emojikeyboard.fragments.FragmentEmojiPeople;
import br.com.instachat.emojikeyboard.fragments.FragmentEmojiPlaces;
import br.com.instachat.emojikeyboard.fragments.FragmentEmojiRecents;
import br.com.instachat.emojikeyboard.fragments.FragmentEmojiSymbols;

public class EmojiTabAdapter extends FragmentPagerAdapter {

    private static FragmentEmojiRecents FRAGMENT_EMOJI_RECENTS = new FragmentEmojiRecents();
    private static FragmentEmojiPeople FRAGMENT_EMOJI_PEOPLE = new FragmentEmojiPeople();
    private static FragmentEmojiNature FRAGMENT_EMOJI_NATURE = new FragmentEmojiNature();
    private static FragmentEmojiObjects FRAGMENT_EMOJI_OBJECTS = new FragmentEmojiObjects();
    private static FragmentEmojiPlaces FRAGMENT_EMOJI_PLACES = new FragmentEmojiPlaces();
    private static FragmentEmojiSymbols FRAGMENT_EMOJI_SYMBOLS = new FragmentEmojiSymbols();


    // CONTRACT
    public EmojiTabAdapter(FragmentManager fm) {
        super(fm);
    }

    // CALLBACKS
    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "RECENTS";
            case 1:
                return "PEOPLE";
            case 2:
                return "NATURE";
            case 3:
                return "OBJECTS";
            case 4:
                return "PLACES";
            case 5:
                return "SYMBOLS";
            default:
                return "UNKNOW";
        }
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        return super.instantiateItem(container, position);
    }

    @Override
    public int getCount() {
        return 6;
    }

    @Override
    public android.support.v4.app.Fragment getItem(int position) {
        switch (position) {
            case 0:
                return FRAGMENT_EMOJI_RECENTS;
            case 1:
                return FRAGMENT_EMOJI_PEOPLE;
            case 2:
                return FRAGMENT_EMOJI_NATURE;
            case 3:
                return FRAGMENT_EMOJI_OBJECTS;
            case 4:
                return FRAGMENT_EMOJI_PLACES;
            case 5:
                return FRAGMENT_EMOJI_SYMBOLS;
            default:
                return FRAGMENT_EMOJI_RECENTS;
        }
    }
}

File: emoji-library/src/main/kotlin/br/com/edsilfer/emojilibrary/view/adapter/EmojiTabAdapter.kt
package br.com.edsilfer.emojilibrary.view.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import br.com.edsilfer.emojilibrary.view.fragments.*
import br.com.edsilfer.emojilibrary.view.listeners.EmojiClickListener

class EmojiTabAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm!!, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val recentEmojis = Recents()
    private val peopleEmojis = People()
    private val natureEmojis = Nature()
    private val objectEmojis = Objects()
    private val placesEmojis = Places()
    private val symbolsEmojis = Symbols()

    init {
        peopleEmojis.registerListener(recentEmojis)
        natureEmojis.registerListener(recentEmojis)
        objectEmojis.registerListener(recentEmojis)
        placesEmojis.registerListener(recentEmojis)
        symbolsEmojis.registerListener(recentEmojis)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> "RECENTS"
            1 -> "PEOPLE"
            2 -> "NATURE"
            3 -> "OBJECTS"
            4 -> "PLACES"
            5 -> "SYMBOLS"
            else -> "UNKNOW"
        }
    }

    override fun getCount(): Int = 6

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> recentEmojis
            1 -> peopleEmojis
            2 -> natureEmojis
            3 -> objectEmojis
            4 -> placesEmojis
            5 -> symbolsEmojis
            else -> recentEmojis
        }
    }

    fun onEmojiClicked(listener: EmojiClickListener) {
        recentEmojis.registerListener(listener)
        peopleEmojis.registerListener(listener)
        natureEmojis.registerListener(listener)
        objectEmojis.registerListener(listener)
        placesEmojis.registerListener(listener)
        symbolsEmojis.registerListener(listener)
    }
}
